### ISSUE TITLE
DB Forge: MySQLi: _process_indexes() is setting the wrong constraint name

### ISSUE THREAD

**Issue by wolf-w** [bcit-ci/CodeIgniter#5388]
Comparing `CI_DB_forge::_process_indexes()` [@1008](https://github.com/bcit-ci/CodeIgniter/blob/develop/system/database/DB_forge.php#L1020) to `CI_DB_mysqli_forge::_process_indexes()` [@235](https://github.com/bcit-ci/CodeIgniter/blob/develop/system/database/drivers/mysqli/mysqli_forge.php#L235)
shows that the index identifier is set different: `$table.'_'.implode('_', $this->keys[$i])` vs. `implode('_', $this->keys[$i])` which leads to that calling `add_key('id')` on two different tables fails with error `Duplicate key name 'id'`.

`CI_DB_mysqli_forge::_process_indexes()` should implement the parent behavior at this point!

---

File abstract:
```php
$sqls[] = 'CREATE INDEX '.$this->db->escape_identifiers($table.'_'.implode('_', $this->keys[$i]))
	.' ON '.$this->db->escape_identifiers($table)
	.' ('.implode(', ', $this->db->escape_identifiers($this->keys[$i])).');';
```
vs.
```php
$sql .= ",\n\tKEY ".$this->db->escape_identifiers(implode('_', $this->keys[$i]))
	.' ('.implode(', ', $this->db->escape_identifiers($this->keys[$i])).')';
```

**Comment by narfbg** [comment]
This is very hard to follow ... I thought you were saying that, under MySQL, using the same `KEY <name>` on 2 different tables would result in an error, but:

```
mysql> create table foo (bar varchar(16) not null, key bar (bar));
Query OK, 0 rows affected (0.03 sec)

mysql> create table bar (bar varchar(16) not null, key bar (bar));
Query OK, 0 rows affected (0.03 sec)

**Comment by wolf-w** [comment]
I tested it again and it turns out, that I set the `id` field accidentally to `unique` as well..  :confounded:

But isn't it generally a cleaner way to prefix the index key with something like `ik_table_`?
Thanks for your help!

**Comment by narfbg** [comment]
What's "cleaner" is subjective.

Prefixing with a key type and table name is more vebose, informative.
Not prefixing makes names shorter and easier to read.

It's mostly a matter of convention, but there isn't one convention to beat them all and we can't satisfy everyone's. Here we err on being shorter, because key names also have *length limits*, and you can easily reach them when you make a composite key and each field name has to be included ...
